### PR TITLE
Various AniList Improvements & Bugfixes

### PR DESCRIPTION
**Pull request by azagon63**
This PR contains a few improvements to the AniList providers:
* Refactor to focus on Media objects, representing parsed responses from the AniList API
* Add more info to metadata (Studios, Tags, Voice Actors, etc, see #49 )
* Fix searching (previously returned 0 or 1 result based on exact match)
* Add a Movie provider (see #64 )

<details><summary>Screenshots</summary>
<p>

AniListSearchResults

</p>
</details>

### CODE REVIEW

**Comment by azagon63** [comment]
Closed and reopened because I hit create before I was done editing 😅

File: Jellyfin.Plugin.Anime/Providers/AniList/AniListApi.cs [hunk 1]
```diff
@@ -16,7 +16,7 @@ namespace Jellyfin.Plugin.Anime.Providers.AniList
     /// Based on the new API from AniList
     /// 🛈 This code works with the API Interface (v2) from AniList
     /// 🛈 https://anilist.gitbooks.io/anilist-apiv2-docs
-    /// 🛈 THIS IS AN UNOFFICAL API INTERFACE FOR EMBY
+    /// 🛈 THIS IS AN UNOFFICAL API INTERFACE FOR JELLYFIN
```
Comment (azagon63) [inline Jellyfin.Plugin.Anime/Providers/AniList/AniListApi.cs:19]: Summary of changes in this file:
* Fixed indentation (mostly in graphql)
* Moved some functions to be methods of the Media object
* Add more attributes to graphql call
* Update comments

File: Jellyfin.Plugin.Anime/Providers/AniList/AniListMovieProvider.cs [hunk 2]
```diff
@@ -0,0 +1,107 @@
+using System;
```
Comment (azagon63) [inline Jellyfin.Plugin.Anime/Providers/AniList/AniListMovieProvider.cs:1]: This file is basically a copy of SeriesProvider with `sed s/Series/Movie/`

File: Jellyfin.Plugin.Anime/Providers/AniList/ApiModel.cs [hunk 3]
```diff
@@ -1,6 +1,12 @@
 ﻿using System;
```
Comment (azagon63) [inline Jellyfin.Plugin.Anime/Providers/AniList/ApiModel.cs:1]: Most of this file is serialization setup for the JSON parser, attributes come from the [AniList API Docs](https://anilist.gitbook.io/anilist-apiv2-docs/) and mostly the [API v2 GraphQL Docs](https://anilist.github.io/ApiV2-GraphQL-Docs/). There's also methods added to convert values from AniList's representation to Jellyfin's.

The idea is you use Media/MediaSearchResult internally (to the plugin) and convert it to the relevant Jellyfin object type through ToMovie and ToSeries, that way both providers can share most of the code.

**Review by crobibero — APPROVED** [review]

File: Jellyfin.Plugin.Anime/Providers/AniList/AniListApi.cs [hunk 4]
```diff
@@ -149,233 +143,72 @@ static AniListApi()
...
-                        return media.id.ToString();
-                    }*/
-                }
-
-                catch (Exception) { }
-            }
-
-            return result;
+            foreach (MediaSearchResult media in WebContent.data.Page.media)
+                return media;
```
Comment (Bond-009) [inline Jellyfin.Plugin.Anime/Providers/AniList/AniListApi.cs:167]: Braces everywhere

File: Jellyfin.Plugin.Anime/Providers/AniList/AniListApi.cs [hunk 5]
```diff
@@ -149,233 +143,72 @@ static AniListApi()
...
-        /// <summary>
-        /// API call to get the description
-        /// </summary>
-        /// <param name="WebContent"></param>
-        /// <returns></returns>
-        public string Get_Overview(RootObject WebContent)
+        public async Task<Media> GetAnime(string id)
         {
-            return WebContent.data.Media.description;
+            return (await WebRequestAPI(AnimeLink.Replace("{0}",id))).data?.Media;
```
Comment (Bond-009) [inline Jellyfin.Plugin.Anime/Providers/AniList/AniListApi.cs]:
```suggestion
            return (await WebRequestAPI(AnimeLink.Replace("{0}", id))).data?.Media;
```

**Review by Bond-009 — CHANGES_REQUESTED** [review]
Missing braces

**Review by oddstr13 — APPROVED** [review]

**Pull request merged**